### PR TITLE
feat: remove token storage, use acquireTokenSilent

### PR DESCRIPTION
**Pull request by Harish-osmosys**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [x] I have performed preliminary unit testing.
- [ ] I have updated the required [api docs](../../apps/api/docs/) as applicable.
- [ ] I have added/updated test cases to the [test suite](../../apps/api/TranscriptSummarizer.postman_collection.json) as applicable

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login endpoint`)
- [x] Description has been added
- [x] Related changes have been added (optional)
- [ ] Screenshots have been added (optional)
- [ ] Query request and response examples have been added (as applicable, in case added or updated)
- [ ] Documentation changes have been listed (as applicable)
- [ ] Test suite output is added (as applicable)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

---

**Description:**

- Use AcquireTokenSilently to get the token for validation 
- Remove Access Token from Session Storage:

**Token Behaviour :**

- The token api is called when we login for the first time 
- Then after for all other operation token api is not called
- When the token expire the token api is called automatically to get the new token 
- The expiry time for one token is Random . Observed values are : 4142 , 4128, 4866 

**Related changes:**

NA

**Screenshots:**

NA

**Query request and response:**

NA

**Documentation changes:**

NA

**Test suite output:**

NA

**Pending actions:**

NA

**Additional notes:**

NA

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved token handling in GraphQL requests to ensure proper authentication and authorization.

- **Chores**
  - Updated import paths for `CurrentUser` decorator in the Summary module.

### CODE REVIEW

File: apps/portal/src/app/graphql/graphql.service.ts [hunk 1]
```diff
@@ -15,8 +17,33 @@
...
+    const token = this.authService.acquireTokenSilent(request).pipe(
+      switchMap((response: AuthenticationResult) => {
+        if (response && response.accessToken) {
+          return [response.accessToken];
+        }
+
+        return throwError('Failed to acquire token');
+      }),
+      catchError((error) => {
+        console.error('Error acquiring token silently', error);
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply apps/portal/src/app/graphql/graphql.service.ts]: removed this line

**Comment by xixas** [comment]
@Harish-osmosys  Add your findings on behaviour of token in description.

File: apps/portal/src/app/graphql/graphql.service.ts [hunk 2]
```diff
@@ -15,8 +17,30 @@ export class GraphqlService {
...
+    };
+    const token = this.authService.acquireTokenSilent(request).pipe(
+      switchMap((response: AuthenticationResult) => {
+        if (response && response.accessToken) {
+          return [response.accessToken];
+        }
+
+        return throwError('Failed to acquire token');
+      }),
+      catchError((error) => throwError(error)),
```
Comment (xixas) [inline apps/portal/src/app/graphql/graphql.service.ts:41]: If it cannot acquire token, do the redirect like we did in app.component.ts(you can test by intentionally failing it.)

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/portal/src/app/graphql/graphql.service.ts (hunk 2 above)
Comment (Harish-osmosys) [reply apps/portal/src/app/graphql/graphql.service.ts]: No it don't redirect if required i can add it

File: apps/portal/src/app/graphql/graphql.service.ts [hunk 3]
```diff
@@ -51,32 +85,32 @@ export class GraphqlService {
...
+      ),
+      catchError((error) => {
+        if (this.isUnauthorizedError(error)) {
+          this.handleUnauthorizedError();
+        }
+
+        return throwError(error);
+      }),
+    );
   }
```
Comment (xixas) [inline apps/portal/src/app/graphql/graphql.service.ts:114]: Make this change
`apps/api/src/Decorator`
Move this to common folder and folder name should be `decorators` casing same as others and plural.

File: apps/portal/src/app/graphql/graphql.service.ts [hunk 4]
```diff
@@ -15,8 +17,38 @@ export class GraphqlService {
...
+          return [response.accessToken];
+        }
+
+        return throwError('Failed to acquire token');
+      }),
+      catchError((error) => {
+        if (error instanceof InteractionRequiredAuthError) {
+          // fallback to interaction when silent call fails
+          this.authService.acquireTokenRedirect(request);
+          return throwError('Redirecting for token acquisition');
```
Comment (xixas) [inline apps/portal/src/app/graphql/graphql.service.ts:45]: Are we throwing this to just show message on UI?
If so can remove it.

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/portal/src/app/graphql/graphql.service.ts (hunk 3 above)
Comment (Harish-osmosys) [reply apps/portal/src/app/graphql/graphql.service.ts:114]: I didn't understand can you please clarify this these changes are done in the portal

Comment (Harish-osmosys) [reply apps/portal/src/app/graphql/graphql.service.ts:113]: Updated as per my understand in this comment : https://github.com/OsmosysSoftware/osm-transcript-summarizer/pull/40/commits/c2f87a6106d0cbdf1ec18c0ef5c8d1303edacbef

File: apps/portal/src/app/graphql/graphql.service.ts (hunk 4 above)
Comment (Harish-osmosys) [reply apps/portal/src/app/graphql/graphql.service.ts]: removed the error throwing line

File: apps/api/src/modules/summary/summary.resolver.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ import { QueryOptionsDto } from 'src/common/graphql/dtos/query-options.dto';
 import { SummaryResponse } from './dto/summary-response.dto';
 import { UseGuards } from '@nestjs/common';
 import { AzureADGuard } from 'src/auth/azure-ad.guard';
-import { CurrentUser } from 'src/Decorator/user.decorator';
+import { CurrentUser } from 'src/common/decorators/user.decorator';
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply apps/api/src/modules/summary/summary.resolver.ts:9]: this is fixed in this comment : https://github.com/OsmosysSoftware/osm-transcript-summarizer/pull/40/commits/07982e85d3cca2a7342cede93f9589349f1fda55

File: apps/portal/src/app/graphql/graphql.service.ts [hunk 6]
```diff
@@ -26,19 +57,21 @@ export class GraphqlService {
...
+          context: {
+            headers: {
+              Authorization: `Bearer ${token}`,
+            },
+          },
+        });
 
-    return queryRef.valueChanges.pipe(
+        return queryRef.valueChanges;
+      }),
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply apps/portal/src/app/graphql/graphql.service.ts:74]: we are handling the specific error here for Unauthorized and rest we are throwing with the info

**Review by xixas — APPROVED** [review]

**Pull request merged**